vulnerabilities/tests/test_pysec.py

```diff
@@ -0,0 +1,489 @@
+        assert get_aliases({"aliases": ["CVE-2021-40831"]}) == ["CVE-2021-40831"]
+        self.assertCountEqual(
+            get_aliases({"aliases": ["CVE-2021-40831"], "id": "GHSA-j3f7-7rmc-6wqj"}),
+            [
+                "CVE-2021-40831",
+                "GHSA-j3f7-7rmc-6wqj",
+            ],
+        )
+        self.assertCountEqual(
+            get_aliases(
+                {"aliases": ["CVE-2022-22817", "GHSA-8vj2-vxx3-667w"], "id": "GHSA-j3f7-7rmc-6wqj"}
+            ),
+            ["CVE-2022-22817", "GHSA-8vj2-vxx3-667w", "GHSA-j3f7-7rmc-6wqj"],
+        )
+
+    def test_get_published_date(self):
+        assert get_published_date(
+            {"id": "GHSA-j3f7-7rmc-6wqj", "published": "2022-01-10T14:12:00Z"}
+        ) == datetime.datetime(2022, 1, 10, 14, 12, 0, 0).replace(tzinfo=datetime.timezone.utc)
+        assert get_published_date(
+            {"id": "GHSA-j3f7-7rmc-6wqj", "published": "2022-01-10T14:12:00.44444Z"}
+        ) == datetime.datetime(2022, 1, 10, 14, 12, 0, 444440).replace(tzinfo=datetime.timezone.utc)
+        assert get_published_date({"id": "GHSA-j3f7-7rmc-6wqj"}) is None
+
+    def test_get_severity(self):
+        assert get_severity(
+            {"id": "GHSA-j3f7-7rmc-6wqj", "ecosystem_specific": {"severity": "HIGH"}}
+        ) == [VulnerabilitySeverity(system=SCORING_SYSTEMS["generic_textual"], value="HIGH")]
+        assert get_severity({"id": "PYSEC-2022-10", "ecosystem_specific": {}}) == []
+        assert get_severity({"id": "PYSEC-2022-10"}) == []
+        assert get_severity(
+            {
+                "id": "PYSEC-2022-10",
+                "severity": [
+                    {"type": "CVSS_V3", "score": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:N"}
+                ],
+            }
+        ) == [
+            VulnerabilitySeverity(
+                system=SCORING_SYSTEMS["cvssv3_vector"],
+                value="CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:N",
+            )
+        ]
+
+        assert get_severity(
+            {
+                "id": "PYSEC-2022-10",
+                "severity": [
+                    {"type": "CVSS_V3", "score": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:N"},
+                    {"type": "CVSS_V3", "score": "CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:N"},
+                ],
+            }
+        ) == [
+            VulnerabilitySeverity(
+                system=SCORING_SYSTEMS["cvssv3_vector"],
+                value="CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:N",
+            ),
+            VulnerabilitySeverity(
+                system=SCORING_SYSTEMS["cvssv3_vector"],
+                value="CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:N",
+            ),
+        ]
+
+    def test_get_references(self):
+        assert get_references(
+            {
+                "id": "PYSEC-2022-10",
+                "references": [
+                    {
+                        "type": "FIX",
```

Comment:
Should we also track this "type" as a new model field? If you think so, please create a follow up issue to solve this later.